contrib/storage-mongo/src/test/java/org/apache/drill/exec/store/mongo/MongoTestSuit.java

```diff
@@ -94,7 +94,8 @@ private static void setup() throws Exception {
       configServers.add(crateConfigServerConfig(CONFIG_SERVER_3_PORT));
 
       // creating replicaSets
-      Map<String, List<IMongodConfig>> replicaSets = new HashMap<>();
+      // A TreeMap ensures that the config servers are started first.
+      Map<String, List<IMongodConfig>> replicaSets = new TreeMap<>();
```

Comment:
Does it require `TreeMap` or `LinkedHashMap`? What order needs to be preserved?

Reply:
The config servers need to be launched first. Using a tree map would guarantee that the CONFIG_REPLICA_SET key would be iterated over first, since it is lexicographically first compared to the other replica set names. So when the flapadoodle library iterates over the replica sets it will see CONFIG_REPLICA_SET first and launch the config servers.

I agree a LinkedHashMap is more intuitive though, so I have changed the code to use that instead.